enhancements/observability/cluster-observability-operator-logging-stack.md

```diff
@@ -0,0 +1,405 @@
+
+### Workflow Description
+
+#### Installation Workflow
+
+For the OpenShift Logging administrator using the COO installation under the proposed system:
+
+1. The user goes to the console and creates an `OperatorConfig` resource. In this resource the user providers an OpenShift Logging subscription channel for both dependent operators (e.g. `stable-5.8`).
+2. The COO creates an `OperatorGroup` and `Subscription` for CLO in `openshift-logging` from the catalog source `redhat-operator` and namespace `openshift-marketplace`.
+3. The COO creates an `OperatorGroup` and `Subscription` for LO in `openshift-operators-redhat` from the catalog source `redhat-operator` and namespace `openshift-marketplace`.
+4. The COO tracks the corresponding `ClusterServiceVersions` for CLO and LO until they reach the phase `succeeded`.
+5. The user goes to the console and creates a `LoggingStack` resource in the `openshift-logging` namespace.
+6. The COO creates a `ClusterLogForwarder` (From `observability.openshift.io/v1`) and a `LokiStack` resource in `openshift-logging` which create and manage all logging stack components.
+7. The COO tracks on a high-level when all stack components (i.e. collection, storage, rules) reach the phase `ready`.
+8. The COO creates an `UIPlugin` resource that will create and manage the [logging-view-plugin](logging-view-plugin) plugin.
+9. The COO creates triggers the [korrel8r](korrel8r) reconciliation if and only if a `MonitoringStack` resource or fallback monitoring resources (See [below](#automated-korrel8r-deployment-with-cmo-uwm)) exist.
+
+#### Upgrades Workflow
+
+For the OpenShift Logging administrator using the COO installation under the proposed system:
+
+1. The user edits the `OperatorConfig` resource by selecting a new OpenShift Logging subscription channel (e.g. `stable-5.9`).
+2. The COO propagates the channel change in the `Subscription` for LO.
+3. The COO tracks the newly selected LO `ClusterServiceVersion` for the new channel until it reaches phase `succeeded`.
+4. The COO tracks the storage and rules components until they reach phase `ready`.
+5. The COO propagates the channel change in the `Subscription` for CLO.
+6. The COO tracks the newly selected CLO `ClusterServiceVersion` for the new channel until it reaches phase `succeeded`.
+7. The COO tracks the collection components until they reach phase `ready`.
+
+### API Extensions
+
+#### OperatorConfig
+
+The `OperatorConfig` resource is the entry point to tell COO on which operator installations it should manage as long as one `LoggingStack` is managed too. It is a cluster-scoped resource.
+
+```yaml
+apiVersion: observability.openshift.io/v1alpha1
+kind: OperatorConfig
+metadata:
+  name: coo-config
+spec:
+  subscriptions:
+  - type: logging
```

Comment:
Does there need to be a way to distinguish collection from storage?  Do you envision we will always expect users to need to use operators from the same channel?